rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,101 @@
+# Opsgenie Integration RFD
+## Required approvers
+
+Engineering:
+Product:
+Security:
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with Opsgenie, allowing access requests to show up as alerts in Opsgenie. This plugin will differ from existing plugins by being part of the Teleport binary directly.
+
+## Success criteria
+Users are able to configure teleport to automatically create alerts in Opsgenie from access requests.
+Alerts were chosen over incidents in Opsgenie for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call 
+
+## Configuration UX
+
+The plugin will be configured using a toml file containing the required information to interact with both Teleport access and the Opsgenie API.
+
+```
+[opsgenie]
+api_key = "path/to/key" # File containing Opsgenie API Key
+opsgenie_addr = "example.app.opsgenie.com" # Address of Opsgenie
+severity = "2" # Severity to create Opsgenie alerts with
+```
```

Comment:
I think at least for now we should leave out the ability to run multiple instances in this way. Since it can be run as a standalone application users can always just run it twice that way and pass each instance a different config file.